client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx

```diff
@@ -0,0 +1,69 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import QuerySitePlans from 'components/data/query-site-plans';
+import SectionHeader from 'components/section-header';
+import { isSiteOnPaidPlan } from 'state/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+
+class DisconnectSurvey extends Component {
+	getReasons() {
+		const { isPaidPlan, translate } = this.props;
+
+		const reasons = [
+			{ slug: 'too-difficult', label: translate( 'It was too hard to configure Jetpack' ) },
+			{ slug: 'missing-feature', label: translate( 'This plan didn’t include what I needed' ) },
+		];
+
+		if ( isPaidPlan ) {
+			reasons.push( { slug: 'too-expensive', label: translate( 'This plan is too expensive' ) } );
+		}
+		return reasons;
+	}
+
+	render() {
+		const { translate, siteId, siteSlug } = this.props;
+		const reasons = this.getReasons();
+
+		return (
+			<div className="disconnect-site__survey main">
+				<QuerySitePlans siteId={ siteId } />
+				<SectionHeader
+					label={ translate(
+						'Would you mind sharing why you want to disconnect %(siteName)s from WordPress.com?',
+						{
+							args: { siteName: siteSlug },
+						}
+					) }
+				/>
+				{ reasons.map( ( { label, slug: reasonSlug } ) => (
```

Comment:
**Minor** but there's some relevant discussion around this [here](https://github.com/Automattic/wp-calypso/pull/17696#discussion_r142611451) and that review that could improve clarity here IMO.

Just to bring it over, there's probably limited benefit from the indirection and `map` here. This section could be something like:

```jsx
<CompactCard onClick={ this.logSelection } className="disconnect-site__survey-one">
  { translate( 'It was too hard to configure Jetpack' ) }
</CompactCard>
<CompactCard onClick={ this.logSelection } className="disconnect-site__survey-one">
  { translate( 'This plan didn’t include what I needed' ) }
</CompactCard>
{ ! isFreeJetpackPlan( site.plan ) && (
  <CompactCard onClick={ this.logSelection } className="disconnect-site__survey-one">
    { translate( 'It was too hard to configure Jetpack' ) }
  </CompactCard>
}
```

Not a blocker if you want to move ahead with the current implementation.